reaper.go

```diff
@@ -108,6 +101,10 @@ func (r *reaper) reap() error {
 			continue
 		}
 
+		// Let's delete the entry here since the channel has been
+		// stored by the caller, in order to wait for the exit code.
+		r.deleteExitCodeCh(pid)
+
```

Comment:
close the chan after sent the exit code. (I can't comment on unmodified code, so I comment here)